PeriodogramAnalysis/test_power_density_statistics.py

```diff
@@ -0,0 +1,96 @@
+from typing import Generator, Any
+
+import numpy as np
+import pytest
+from miv.core.datatype import Signal
+from power_density_statistics import (
+    SpectrumAnalysisBase,
+    SpectrumAnalysisPeriodogram,
+    SpectrumAnalysisWelch,
+)
+
+
+def mock_signal_generator():
+    timestamps = np.linspace(0, 10, 1000, endpoint=False)
+    signal = (
+        np.sin(2 * np.pi * 5 * timestamps)
+        + np.sin(2 * np.pi * 10 * timestamps)
+        + np.sin(2 * np.pi * 15 * timestamps)
+        + np.sin(2 * np.pi * 20 * timestamps)
+    )
+
+    data1 = np.array([signal, signal]).T
+    signal = Signal(data=data1, timestamps=timestamps, rate=100)
+    return signal
+
+
+def test_call_invalid_input_parameters():
+    with pytest.raises(NotImplementedError):
+        analyzer = SpectrumAnalysisBase(window_length_for_welch=4)
+        analyzer(mock_signal_generator())
+
+    with pytest.raises(ValueError):
+        SpectrumAnalysisWelch(window_length_for_welch=-1)
+
+    with pytest.raises(ValueError):
+        SpectrumAnalysisWelch(window_length_for_welch=0.75)
+
+
+def test_call_return_shape():
+    analyzer = SpectrumAnalysisWelch(window_length_for_welch=4)
+    result = analyzer(mock_signal_generator())
+    freqs, psd = result
+
+    assert isinstance(result, tuple)
+    assert len(result) == 2
+    assert isinstance(freqs, np.ndarray)
+    assert isinstance(psd, np.ndarray)
+    assert freqs.shape[0] == 801
```

Comment:
Where is this number coming from??

Reply:
It is calculated based on window length, sampling rate and FFT nodes.
\[
\Delta f = \frac{100}{1600} = 0.0625 \, \text{Hz}
\]